frontend/components/common/ProductGrid.tsx

```diff
@@ -19,9 +19,19 @@ export function ProductGrid({
 }: ProductGridProps) {
    const childrenCount = React.Children.count(children);
 
-   const visibleChildrenCount = useBreakpointValue(
+   const columnsToGetAtLeastOneCompleteRow: Record<string, number> =
       Object.entries(columns).reduce((acc, [key, value]) => {
-         acc[key] = Math.floor(childrenCount / value) * value;
+         acc[key] = childrenCount < value ? childrenCount : value;
```

Comment:
Super minor: I might find `Math.min` easier to read here

```suggestion
         acc[key] = Math.min(childrenCount, value);
```